### PR TITLE
[Debug][HttpKernel] remove frames added by DebugClassLoader in stack traces

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #28818
| License       | MIT
| Doc PR        | -

image

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Kernel.php [hunk 1]
```diff
@@ -504,21 +505,27 @@ protected function initializeContainer()
...
-                $backtrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 3);
+                $backtrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 5);
                 // Clean the trace by removing first frames added by the error handler itself.
                 for ($i = 0; isset($backtrace[$i]); ++$i) {
                     if (isset($backtrace[$i]['file'], $backtrace[$i]['line']) && $backtrace[$i]['line'] === $line && $backtrace[$i]['file'] === $file) {
                         $backtrace = \array_slice($backtrace, 1 + $i);
                         break;
                     }
                 }
+                for ($i = \count($backtrace) - 2; 0 < $i; --$i) {
```
Comment (stof) [inline src/Symfony/Component/HttpKernel/Kernel.php]: I would add the explanation comment before this loop, as done for the previous one.

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpKernel/Kernel.php]: comment added

**Review by j4nr6n — APPROVED** [review]
Tested and working in my scratch project. Looks really good. :+1: 
image

**Comment by Tobion** [comment]
@j4nr6n the example you used is not affected by this change. This is about the deprecations coming from the DebugClassLoader which is not triggered in your example.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**